src/lib/Actors/ActorProxy.php

```diff
@@ -29,22 +30,29 @@ public function __construct(protected ProxyFactory $proxyFactory, protected Logg
      * Returns an actor proxy
      *
      * @param string $interface
-     * @param mixed $id The id to proxy for
+     * @param string|IActorReference $id_or_reference The id or actor reference to proxy for
      * @param string|null $override_type Allow overriding the Dapr type for a given interface
      *
      * @return object
-     * @throws ReflectionException
      * @throws DependencyException
      * @throws NotFoundException
+     * @throws ReflectionException
      */
-    public function get(string $interface, mixed $id, string|null $override_type = null): object
-    {
-        $this->logger?->debug('Getting actor proxy for {i}||{id}', ['i' => $interface, 'id' => $id]);
+    public function get(
+        string $interface,
```

Comment:
Also, it is odd that interface can be passed along with the new actor reference - it can be conflicting.

Reply:
Hmm. Yeah, the reference could hold the interface type as well. My only issue with that would be when code changes would require a different interface; updating references might be complicated.

Reply:
I should also add that there's no mapping between a dapr type and an interface; there's only the attribute on the interface. In PHP, there's no way to scan for existing attributes like in other languages because it's possible the file containing the attribute hasn't been loaded into memory yet.